features/networking/egress-ip.feature

```diff
@@ -232,3 +232,127 @@ Feature: Egress IP related features
+    And the output should contain "<%= cb.valid_ip %>"
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-15998
+  @admin
+  Scenario Outline: Invalid egressIP should not be acceptable
+    Given I select a random node's host
+    When I run the :patch admin command with:
+      | resource      | hostsubnet                     |
+      | resource_name | <%= node.name %>               |
+      | p             | {"egressIPs": <invalid_ip> }   |
+      | type          | merge                          |
+    Then the step should fail
+    And the output should contain "Invalid JSON Patch"
+
+    Examples:
+      | invalid_ip              |
+      | fe80::5054:ff:fedd:3698 |
+      | a.b.c.d                 |
+      | 10.10.10.-1             |
+      | 10.0.0.1/64             |
+      | 10.1.1/24               |
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-25694
+  @admin
+  @destructive
+  Scenario: Random outages with egressIP
+    Given I store the masters in the :nodes clipboard
+    And the valid egress IP is added to the "<%= cb.nodes[0].name %>" node
+    And I have a project
+    And I have a pod-for-ping in the project
+
+    # add the egress ip to the project
+    Given as admin I successfully merge patch resource "netnamespace/<%= project.name %>" with:
+      | {"egressIPs": ["<%= cb.valid_ip %>"]} |
+
+    # Access some external host that is not responding
+    When I execute on the pod:
+      | nc | -v | -w2 | 8.8.8.8 | 23 |
+    Then the step should fail
+    And the output should contain "Operation timed out"
+
+    #Check the sdn logs on pod node
+    Given I use the "<%= pod.node_name %>" node
+    When I get the networking components logs of the node since "90s" ago
+    And the output should not contain "may be offline"
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-25640
+  @admin
+  @destructive
+  Scenario: Should be able to access to the service's externalIP with egressIP
+    Given I have a project
+    And SCC "privileged" is added to the "system:serviceaccounts:<%= project.name %>" group
+    Given I store the schedulable nodes in the :nodes clipboard
+    And the Internal IP of node "<%= cb.nodes[0].name %>" is stored in the :hostip clipboard
+
+    # Create additional network through CNO
+    Given as admin I successfully merge patch resource "networks.config.openshift.io/cluster" with:
+      | {"spec":{"externalIP":{"policy":{"allowedCIDRs":["<%= cb.hostip %>/24"]}}}} |
+
+    # Clean-up required to erase above externalIP policy after testing done
+    Given I register clean-up steps:
+    """
+    Given as admin I successfully merge patch resource "networks.config.openshift.io/cluster" with:
+      | {"spec":{"externalIP":{"policy":{"allowedCIDRs": null}}}} |
+    """
+
+    # Create a svc with externalIP
+    Given I switch to the first user
+    Given I obtain test data file "networking/externalip_service1.json"
+    And I wait up to 300 seconds for the steps to pass:
```

Comment:
need to wait up 300 seconds for creating svc?

Reply:
@weliang1  There is a bug, right?

Reply:
@huiran0826 yes, the bug is not fixed yet: https://bugzilla.redhat.com/show_bug.cgi?id=1793099

Reply:
@weliang1  Thanks!

Reply:
We are hiding bugs in this way.  We should let it fail given there are bugs affecting this scenario.

Reply:
Agree with you! We should not hide the bug.
But actually from the comments in bug https://bugzilla.redhat.com/show_bug.cgi?id=1793099, it is still not decided it is a valid bug or the time is reasonable.   I think we can keep the time here for now and for externalIP service scenario, there are other cases tracked to test it.

For this case it mainly to test egressIP which also has one bug open  which cause the case failing and we did not workaround it just let it fail.

Reply:
@huiran0826 after checking your steps again, you do not need line 331 to sleep 300 seconds.  We will hit bug #1793099 only when running two externalIP automation cases together, the first case do not need sleep 300s, but the second case will update externalIP policy to use new allow/deny CIDR, this updating process right now need at least 220 seconds to accept the new creating service.
For your testing here, you only has one externalIP configuration, I assume without sleeping 300 seconds, this case can still pass

Reply:
Thanks @weliang1,  any suggestion what time I should use?

Reply:
@huiran0826 The deny/allow policy for externalIP is for cluster, but the svc configuration is for project. Agree, we do not know if other externalIP cases will run before this case.  We use 300 seconds to continue creating the svc until creation pass, if now other externalIP case before your case, your svc creation will success on first try. You need to keep this step.

Reply:
Thanks!